service/src/test/java/bio/terra/catalog/service/BaseStatusServiceTest.java

```diff
@@ -23,4 +26,23 @@ void getCurrentStatus() {
         service.getCurrentStatus(),
         is(new SystemStatus().ok(true).systems(Map.of("test", status))));
   }
+
+  interface Status extends Supplier<SystemStatusSystems> {}
+
+  @Test
+  void startStatusChecking() {
+    var config = new StatusCheckConfiguration(true, 1, 0, 10);
+    BaseStatusService service = new BaseStatusService(config);
+    var status = mock(Status.class);
+    service.registerStatusCheck("", status);
+    service.startStatusChecking();
+    verify(status).get();
+  }
+
+  @Test
+  void getNonEnabledStatus() {
+    var config = new StatusCheckConfiguration(false, 0, 0, 10);
+    BaseStatusService service = new BaseStatusService(config);
+    assertThat(service.getCurrentStatus(), is(new SystemStatus().ok(true)));
```

Comment:
Why does this show a green status when the test is `NonEnabled`?

Reply:
well in the method weather or not the status is enabled or not is still returns a green status

Reply:
It returns `true`, although I wonder if it should? Maybe we should be returning `false` instead. I'm not even sure why the configuration attribute `enabled` exists in the first place, it's always `true` in all our configurations.

Reply:
Resolving this is probably out of scope for this PR. Keep the test you have and create a tech debt ticket in JIRA so we can look at this later.